docs/usage/configuration.md

```diff
@@ -0,0 +1,216 @@
+---
+jupytext:
+  cell_metadata_filter: -all
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.12
+    jupytext_version: 1.7.1
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+# Configuration
+
+If used without arguments, the {func}`~magicgui.magicgui` decorator will build
+you a GUI using the default settings, making some reasonable default guesses
+about [what type of GUI widget to use](types_widgets), based on the type or
+type annotation of your argument:
+
+```{code-cell} python
+import enum
+import pathlib
+
+from datetime import datetime
+from magicgui import magicgui
+
+class Choice(enum.Enum):
+    A = 'Choice A'
+    B = 'Choice B'
+    C = 'Choice C'
+
+@magicgui
+def widget_demo(
+    boolean=True,
+    number=1,
+    string="Text goes here",
+    dropdown=Choice.A,
+    filename=pathlib.Path.home(),
+):
+    """Run some computation."""
+    ...
+
+widget_demo.show()
```

Comment:
Fun facts from a separate conversation with @tlambert03: this returns the widget, which has a [`_repr_png_` method](https://github.com/napari/magicgui/blob/9211da912a700d6a4680cb4d6ed89637fd9df135/magicgui/widgets/_bases.py#L321), which means jupyter book is able to render it in the docs. BUT, the Qt app is not being `exec_`d. So the Qt event loop does not actually need to be running to display things, only for interactivity. This has rather far-reaching consequences!

Reply:
wow, yes, very interesting!!!